core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -342,6 +342,13 @@ public final double readDouble() throws IOException {
         return Double.longBitsToDouble(readLong());
     }
 
+    public final Double readOptionalDouble() throws IOException {
```

Comment:
no I mean that I had missed that you had tests in there....I commented too early, sorry, it is fine as-is! I wonder if we have missed to add tests to BytesStreamTests for similar changes